lib/reek/configuration/configuration_converter.rb

```diff
@@ -101,10 +103,25 @@ def strings_to_regexes_for_directories
                   to_regex item
                 end
               end
+            rescue NoMethodError
+              warn_about_missing_configuration(detector) if configuration.nil?
             end
           end
         end
       end
+
+      def warn_about_missing_configuration(detector)
```

Comment:
The way error handling works in Reek is that we raise specific errors throughout the code and then do the warning in `lib/reek/cli/application.rb`.

We do this, among other reasons, because it allows us to test the error handling code without having warnings appear all over the place during the spec run.

In this case, the error should probably be a `Reek::Errors::ConfigFileError`. It will be handled here: https://github.com/troessner/reek/blob/b1fa495b4f0fa617552a71790e2fed56db667f77/lib/reek/cli/application.rb#L48